components/cdc/src/endpoint.rs

```diff
@@ -1044,58 +1015,86 @@ struct Initializer {
 }
 
 impl Initializer {
-    async fn on_change_cmd(&mut self, mut resp: ReadResponse<RocksSnapshot>) {
-        CDC_SCAN_TASKS.with_label_values(&["total"]).inc();
-        let deregister = if let Some(region_snapshot) = resp.snapshot {
+    async fn initialize<T: 'static + RaftStoreRouter<RocksEngine>>(
+        &mut self,
+        change_cmd: ChangeObserver,
+        raft_router: T,
+        concurrency_semaphore: Arc<Semaphore>,
+    ) -> Result<()> {
+        let _permit = concurrency_semaphore.acquire().await;
+
+        // The initialization maybe canceled after acquiring a permit.
+        if self.downstream_state.load() == DownstreamState::Stopped {
```

Comment:
Can you add a comment on where `downstream_state` is written to?

Reply:
Yes, in the current implementation it has be to true. It's false only in tests that do not care about the barrier. This parameter can be removed after the segmentation scan is supported.